src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,229 @@
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete car by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT * FROM cars AS c JOIN cars_drivers AS cd "
+                + "ON c.id = cd.car_id "
+                + "WHERE driver_id = ? AND c.is_deleted = false;";
+        List<Car> carsByDriver = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Car car = new Car();
+                car.setId(resultSet.getObject("id", Long.class));
+                car.setModel(resultSet.getString("model"));
+                Manufacturer manufacturer = new Manufacturer();
+                manufacturer.setId(resultSet.getObject("manufacturer_id", Long.class));
+                car.setManufacturer(manufacturer);
+                carsByDriver.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all cars by driver id: " + driverId, e);
+        }
+        addDriversToCarList(carsByDriver);
+        addInfoAboutManufacturer(carsByDriver);
+        return carsByDriver;
+    }
+
+    private void addInfoAboutManufacturer(List<Car> cars) {
+        String query = "SELECT name, country FROM manufacturers "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            for (Car oneCar : cars) {
+                statement.setLong(1, oneCar.getManufacturer().getId());
+                ResultSet resultSet = statement.executeQuery();
+                if (resultSet.next()) {
+                    Manufacturer manufacturer = oneCar.getManufacturer();
+                    manufacturer.setName(resultSet.getString("name"));
+                    manufacturer.setCountry(resultSet.getString("country"));
+                    oneCar.setManufacturer(manufacturer);
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get manufacturer for cars: " + cars, e);
+        }
+    }
+
+    private void addDriversToCarList(List<Car> cars) {
+        if (!cars.isEmpty()) {
```

Comment:
remove this check, remove the method, and move for loop in places where it needs